### ISSUE TITLE
class-file injection not working for 1.8.16 on Java 11-ea+25 

### ISSUE THREAD

**Issue by snazy** [raphw/byte-buddy#507]
`net.bytebuddy.dynamic.loading.ClassInjector.UsingReflection.Dispatcher.CreationAction#run` doesn't try to run `Dispatcher.UsingUnsafeOverride.make()` instead of `Dispatcher.UsingUnsafeInjection.make()`. It seems, that the catch-clause in `net.bytebuddy.dynamic.loading.ClassInjector.UsingReflection.Dispatcher.CreationAction#run` needs to catch `IllegalStateException` in addition to `NoSuchMethodException`, or just `Exception`.

```
java.lang.IllegalStateException: Could not find sun.misc.Unsafe
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingUnsafe$Dispatcher$Disabled.initialize(ClassInjector.java:1728)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingUnsafe.inject(ClassInjector.java:1564)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$ForUnsafeInjection.load(ClassLoadingStrategy.java:495)
	at net.bytebuddy.dynamic.TypeResolutionStrategy$Passive.initialize(TypeResolutionStrategy.java:79)
	at net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:4247)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$UsingUnsafeInjection.make(ClassInjector.java:755)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$CreationAction.run(ClassInjector.java:375)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection$Dispatcher$CreationAction.run(ClassInjector.java:362)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingReflection.<clinit>(ClassInjector.java:68)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$Default$InjectionDispatcher.load(ClassLoadingStrategy.java:202)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$Default.load(ClassLoadingStrategy.java:126)
	at net.bytebuddy.dynamic.TypeResolutionStrategy$Passive.initialize(TypeResolutionStrategy.java:79)
	at net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:4247)
```

**Comment by snazy** [comment]
After broadening the catch clause to catch `Exception`, the same code runs fine.

**Comment by snazy** [comment]
If it helps, the failure happens with a code like 
```
ByteBuddy()
    .subclass(SomeClass.class)
    .make()
    .load(Thread.currentThread().getContextClassLoader(), ClassLoadingStrategy.Default.INJECTION)
    .getLoaded()
```

**Comment by raphw** [comment]
Fixed on master, thanks for reporting.

**Comment by snazy** [comment]
Note: fixed with 1838799b47e4b4b5b84128884ac8a1ff86f5cf45
Works fine! Thanks!
